openapi/data_repository_service.swagger.yaml

```diff
@@ -319,13 +325,22 @@ definitions:
     properties:
       object:
         $ref: '#/definitions/Object'
-  GetAccessURLResponse:
+  AccessURL:
     type: object
-    required: ['object']
+    required: ['url']
     properties:
-      access_url:
+      url:
         type: string
-        description: A fully resolvable access_url that can be used to fetch the actual object bytes.
+        description: A fully resolvable URL that can be used to fetch the actual object bytes.
+      headers:
+        type: array
+        items:
+          type: string
+        description: >-
+          An optional list of headers to include in the HTTP request to `url`.
```

Comment:
I think this is a great first step.  It improves what we have right now and we should go for it.  As an implementer/user of DRS I'd like to know 1) what are the possible headers here, do we define them in the /service-info end point for example or in the spec as a list of acceptable ones?  And 2) it would be great to see examples of how these work for each of the types of access methods DRS 1.0 would support (in the docs for example).  Totally fine to address in a future PR

Reply:
Re (1): As a user, you shouldn't have to know what headers are possible; you just use the ones you're given. As an implementor, you can return any headers you want. The most likely use case is to return an Authorization header, as is shown in the example in the YAML.

Re (2): this PR already includes beefed-up documentation on access methods (in front_matter.adoc); happy to add more in future PRs